README.md

```diff
@@ -43,32 +43,34 @@ for definition of maturity levels).
 
 Components marked `Stable` provide the following guarantees:
 
-- Field types will not change.
-- Field numbers will not change.
-- Numbers assigned to enum choices will not change.
-- Service names and service package names will not change.
-- Service operation names, parameter and return types will not change.
+- Field types, numbers and names will not change.
+- Names of messages and enums will not change.
+- Names of enum choices and numbers assigned to enum choices will not change.
+- The location of messages and enums, i.e. whether they are declared at the top lexical
+  scope or nested inside another message will not change.
+- Package names and directory structure will not change.
+- gRPC service names, service method names, parameters and return types will not change.
+- Service method parameter names will not change.
+- Service method kind (unary vs streaming) will not change.
+- No existing symbol will be deleted.
+- `optional` and `repeated` declarators of existing fields will not change.
```

Comment:
Why for `optional`? Changing from non-optional to optional should be fine, even the other way if the default value is ok.

Reply:
Changing from optional to non-optional can break the generated code since it affects the presence check functions.